### PR TITLE
chore: treeshake load function code if we know it's unused

### PR DESCRIPTION
**Pull request by dummdidumm**
This shaves off about ~1kb minified when you're not using any load functions in your code base, which can become more common with load functions

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by Rich-Harris** [comment]
nice! The test failures look legit though. Also: what happens if I do `export * from '$lib/common'` or something? Obviously a real edge case but do we de-opt in that scenario?

**Comment by dummdidumm** [comment]
We deopt in the static analysis in that case, yes

File: packages/kit/src/exports/vite/index.js [hunk 1]
```diff
@@ -369,12 +369,27 @@ async function kit({ svelte_config }) {
...
+					// Through the finished analysis we can now check if any node has server or universal load functions
+					const has_server_load = build_metadata
+						? Object.values(build_metadata.nodes).some((node) => node.has_server_load)
+						: true;
+					const has_universal_load = build_metadata
+						? Object.values(build_metadata.nodes).some((node) => node.has_universal_load)
+						: true;
+					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = s(has_server_load);
+					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = s(has_universal_load);
 				}
```
Comment (Rich-Harris) [inline packages/kit/src/exports/vite/index.js:386]: `build_metadata` is guaranteed to exist at this point, so rather than having the pair of ternaries (which make it seem like there are more possibilities than there actually are) what if we do this, using `build_metadata` as a proxy for `secondary_build_started`?

```suggestion
				if (build_metadata) {
					const nodes = Object.values(build_metadata.nodes);

					// Through the finished analysis we can now check if any node has server or universal load functions
					const has_server_load = nodes.some((node) => node.has_server_load);
					const has_universal_load = nodes.some((node) => node.has_universal_load);

					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = s(has_server_load);
					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = s(has_universal_load);
				} else {
					manifest_data = sync.all(svelte_config, config_env.mode).manifest_data;
					
					// During the initial server build we don't know yet
					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = 'true';
					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = 'true';
				}
```

**Review by Rich-Harris — APPROVED** [review]
one small suggestion but LGTM, this is smart

File: packages/kit/src/exports/vite/index.js (hunk 1 above)
Comment (dummdidumm) [reply packages/kit/src/exports/vite/index.js:386]: I like the idea of making it not a ternary but I'd prefer a type cast instead (that will also fail loudly if something goes wrong down the line)
